Framework/CurveFitting/src/Algorithms/VesuvioCalculateGammaBackground.cpp

```diff
@@ -542,26 +540,30 @@ void VesuvioCalculateGammaBackground::cacheInstrumentGeometry() {
   m_foils1.resize(nfoils);
   for (size_t i = 0; i < nfoils; ++i) {
     const auto &foil0 = foils0[i];
+    if (!foil0)
+      throw std::invalid_argument("Error - foil component is NULL.");
```

Comment:
Checking above:
```cpp
auto foils0 = inst->getAllComponentsWithName("foil-pos0");
auto foils1 = inst->getAllComponentsWithName("foil-pos1");
```
I think `Instrument` does not return `nullptr` in this vectors, so the check is not necessary?

No, there is no way of obtaining a shared pointer to a detector, since we intend to remove that mechanism altogether.